src/hooks/use-proposals.js

```diff
@@ -97,8 +98,37 @@ const useProposals = () => {
 
       if (!oldProposal) return _proposal
 
+      const totalVotingPnt = _proposal.yes.plus(_proposal.no).plus(oldProposal.yes).plus(oldProposal.no)
+      const totalYes = _proposal.yes.plus(oldProposal.yes)
+      const totalNo = _proposal.no.plus(oldProposal.no)
+      const percentageYea = totalYes.dividedBy(totalVotingPnt).multipliedBy(100)
+      const percentageNay = totalNo.dividedBy(totalVotingPnt).multipliedBy(100)
+      const totalYesPercentage = formatAssetAmount(percentageYea, '%', { decimals: 2 })
+      const totalNoPercentage = formatAssetAmount(percentageNay, '%', { decimals: 2 })
+
+      const minAcceptQuorum =
+        _proposal.minAcceptQuorum > oldProposal.minAcceptQuorum
+          ? _proposal.minAcceptQuorum
+          : oldProposal.minAcceptQuorum
+
+      const totalVotingPower = _proposal.votingPower.plus(oldProposal.votingPower)
+      const totalQuorum = _proposal.yes.plus(oldProposal.yes).dividedBy(totalVotingPower)
+      const biggestMinQuorum = minAcceptQuorum
+      const totalQuorumReached = totalQuorum.isGreaterThan(biggestMinQuorum)
+      const totalPassed = percentageYea.isGreaterThan(51) && totalQuorumReached
+
+      const totalOpen = _proposal.open || oldProposal.open
+
+      _proposal.formattedPercentageYea = totalYesPercentage
```

Comment:
Here it's better to create a new object
or set the attributes in the return value after the `..._proposal`.